src/linker/Linker.Steps/LinkAttributesStep.cs

```diff
@@ -36,9 +36,22 @@ IEnumerable<CustomAttribute> ProcessAttributes (XPathNavigator nav, ICustomAttri
 				TypeDefinition attributeType;
 
 				string internalAttribute = GetAttribute (iterator.Current, "internal");
-				if (internalAttribute == "RemoveAttributeInstances" && provider != null && !Annotations.IsMarked (provider)) {
-					IEnumerable<Attribute> removeAttributeInstance = new List<Attribute> { new RemoveAttributeInstancesAttribute () };
-					Context.CustomAttributes.AddInternalAttributes (provider, removeAttributeInstance);
+				if (internalAttribute != String.Empty) {
+					switch (internalAttribute) {
+					case "RemoveAttributeInstances":
+						if (provider != null) {
```

Comment:
I think I don't understand why `provider != null` implies that this is a type.

Reply:
ProcessAttributes function is only called with a provider when it's called inside ProcessType method, all the other methods don't pass a provider and ProcessAttributes function has a provider by default in null. But I guess I will need to change that if I want to print which member type was being used.